forte/data/base_pack.py

```diff
@@ -97,6 +98,7 @@ def __init__(self, pack_name: Optional[str] = None):
         super().__init__()
         self.links: List[LinkType] = []
         self.groups: List[GroupType] = []
+        self.pack_version: List[int] = PACK_VERSION
```

Comment:
We add some explanations for `pack_version` in the docstring, and explain how this will affect serialization and what kind of exceptions would be thrown.